docs/dev/convert_layout.rst

```diff
@@ -116,16 +117,21 @@ These steps happen for each operator in sequence, where ConvertLayout pass keeps
         data_layout = attrs['data_layout']
         kernel_layout = attrs['kernel_layout']
         data, weight = inputs
-        assert desired_layout == 'NCHW', \
-                "Currently only transformation to NCHW layout is supported."
+
+        # Use the first entry in desired layouts which specifies the data layout.
+        # The expected ordering of layouts for this operator is defined by this function.
+        desired_data_layout = str(desired_layouts[0])
+
         if desired_layout == 'NCHW':
             new_attrs = dict(attrs)
-            new_attrs['data_layout'] = desired_layout
+            new_attrs['data_layout'] = desired_data_layout
             new_attrs['kernel_layout'] = 'OIHW'
             # Actual insertion of layout transforms is taken care internally
             # by ConvertLayout pass.
             return relay.nn.conv2d(data, weight, **new_attrs)
-        return None
+        else:
+            assert "Layout %s is not yet supported" % desired_data_layout
+            return None
```

Comment:
This return is unreachable.

Reply:
One way to do this is to move the assert to L124.

~~~
assert desired_layout == ...
if desired_layout == 'NCHW':
    .....
return None
~~~

Reply:
I think this goes back to @comaniac's original comment about unnecessary branching after an assert, I think just removing the unnecessary return is enough?

Reply:
I'm fine with either proposals I mentioned before, but the one without an if-block might be better considering @anijain2305's comment. In this case, you don't have to return None naturally.

```
assert layout == 'NCHW'
# Do something.
return relay.nn.conv2d(data, weight, **new_attrs)
```

Reply:
I am fine with either way.

Reply:
I'll go with the assert, thanks